lib/index.js

```diff
@@ -114,7 +116,7 @@ module.exports = function ({
       return hash.match(/^sha1$|^sha256$/)
     }) || hashes[hashes.length - 1]
   )
-  .update((new Date()).valueOf().toString())
+  .update(sessionAuthorizationKeySecret || (new Date()).valueOf().toString())
```

Comment:
Since we get `valueOf()` and convert `toString()`, maybe we could allow for object type for the parameter; so IRT the jsdoc comment above:
> * @param {string|object} [param0.authorizationId] Identifier for the authorization state saved in the session object, so that an authorization could be shared between the helper instances and used for horizontal scaling.

Reply:
applying the methods `valueOf()` and `toString()` consecutively will not properly work with objects, as this will always resolve into `'[object Object]'`. We could stringily the it, assuming it to be a valide JSON object with:
`.update(JSON.stringify(authorizationId) || (new Date()).valueOf().toString())`

But I don't see value in allowing the authorizationId parameter to be an object. Where would you see the benefits?

Reply:
TL;DR, please never mind, I misread parenthesis in your change.

On a side note, the only potential benefit I can think of for allowing an arbitrary input is letting developers obtain their `authorizationId` automatically, from a service of some kind. But you are right, handling it would require an unnecessary complication of the library code or assumptions. It is cleaner to require `{string}`.